src/ucs/debug/debug.c

```diff
@@ -1331,13 +1331,12 @@ void ucs_debug_disable_signals()
     ucs_recursive_spin_unlock(&ucs_kh_lock);
 }
 
-
+#ifdef __SANITIZE_ADDRESS__
```

Comment:
after this change, if UCX was built without ASAN, and the user program is using ASAN, it would call an undefined symbol ucs_debug_asan_validate_address.
lets keep only the check itself under ifdef